docs/configuration/configuration.md

```diff
@@ -1,87 +1,143 @@
-| providerID  | optional `string`                                          | Value binds to grpc connection's providerID field. GRPC server implementations may use this to identify connecting flagd instance |
-| selector    | optional `string`                                          | Value binds to grpc connection's selector field. GRPC server implementations may use this to filter flag configurations           |
-| certPath    | optional `string`                                          | Used for grpcs sync when TLS certificate is needed. If not provided, system certificates will be used for TLS connection          |
-
-The `uri` field values do not need to follow the [URI patterns](#uri-patterns), the provider type is instead derived from the provider field.
-If the prefix is supplied, it will be removed on startup without error.
+
+The flagd accepts a string argument, which should be a JSON representation of an array of `SourceConfig` objects.
+
+Alternatively, these configurations should be passed to flagd via config file, specified using the `--config` flag.
+
+| Field       | Type               | Note                                                                                                                                         |
+|-------------|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| uri         | required `string`  | Flag configuration source of the provider                                                                                                    |
+| provider    | required `string`  | Provider type - `file`, `kubernetes`, `http` or `grpc`                                                                                       |
+| bearerToken | optional `string`  | Used for http sync and token get appended to `Authorization` header with [bearer schema](https://www.rfc-editor.org/rfc/rfc6750#section-2.1) |
+| providerID  | optional `string`  | Value binds to grpc connection's providerID field. GRPC server implementations may use this to identify connecting flagd instance            |
+| selector    | optional `string`  | Value binds to grpc connection's selector field. GRPC server implementations may use this to filter flag configurations                      |
+| grpcSecure  | optional `boolean` | Used to enable secure TLS connectivity for grpc sync. Default(ex:- if unset) is false, which will use insecure grpc connection               |
+| certPath    | optional `string`  | Used for grpcs sync when TLS certificate is needed. If not provided, system certificates will be used for TLS connection                     |
+
+The `uri` field values **do not** need to follow the [URI patterns](#uri-patterns), the provider type is instead derived
```

Comment:
the phrasing here could be stronger, 'do not need' feels as though it would still work, whereas flagd would error if you do

Reply:
Thanks, yes will update this :)

Reply:
Updated the documentation to emphasize the difference between uri patterns and source configuration URI